### ISSUE TITLE
issue about edit metadata ( V1.0.0)

### ISSUE THREAD

**Issue by webb-lu** [sitewhere/sitewhere#4]
hi, I meet some problem when I try to edit the metadata. 
When I create a device with metadata, then edit it. I found that I cannot remove all the metadata.

for example.
create device A with metadata like :  { "key1" : "value1" }
then I try to remove all the metadata, so I send a empty json object as the metadata obj in the Restful API like : {}
then the data in the database won't be changed. => still  { "key1" : "value1" }

but if I send another set of key value pair like  { "key2" : "value2" }
then I will get the correct result => { "key2" : "value2" }
It's seems the metadata json object must keep at least one pair inside.

this issue was not just occur in the edit device pages, including the edit user , device group and specification pages all have the same problem. Especially the device group, it cannot remove any metadata. 

Could you help to check this issue ? thanks.
( but this issue was not occur on edit site page )

By the way, does the feature of creating a device with a specific site token was done? 
https://github.com/sitewhere/sitewhere-android/issues/3

Thanks again~~

**Comment by derekadams** [comment]
We verified the bug. You are correct that passing an empty JSON array as the metadata in updates does not work. It appears to be a bug in the common update logic. I've added two Jira issues for the 1.0.2 release so that we make sure the problems are taken care of. Thanks for reporting the problem!

https://sitewhere.atlassian.net/browse/SITEWHERE-145
https://sitewhere.atlassian.net/browse/SITEWHERE-146
